src/algebra/big_operators/basic.lean

```diff
@@ -817,6 +817,15 @@ lemma prod_multiset_count [decidable_eq α] [comm_monoid α] (s : multiset α) :
   s.prod = ∏ m in s.to_finset, m ^ (s.count m) :=
 by { convert prod_multiset_map_count s id, rw map_id }
 
+@[to_additive] lemma prod_mem_multiset {α M : Type*} [decidable_eq α]
+  (m : multiset α) (f : {x // x ∈ m} → β) (g : α → β)
```

Comment:
In the end I decided to make `f` and `g` different, although I could also specify `f = g \comp subtype.val`, so that `f` is allowed to depend on the proof that `x ∈ m`. Otherwise you get some annoying messing about with `prod_attach`.